mod/caching/set/all/contributions.rb

```diff
@@ -1,63 +1,75 @@
 def update_direct_contribution_count
   return unless respond_to? :direct_contribution_count
+  new_count = intrusive_family_acts.count
+  update_contribution_count_card direct_contribution_count_card, new_count
+end
 
-  new_contr_count = intrusive_family_acts.count
-  Card::Auth.as_bot do
-    if direct_contribution_count_card.new_card?
-      direct_contribution_count_card.update_attributes!(:content => new_contr_count.to_s)
-    else
-      direct_contribution_count_card.update_column(:db_content, new_contr_count.to_s)
-      direct_contribution_count_card.expire
-    end
+def calculate_direct_contribution_count
+  if respond_to? :direct_contribution_count
+    direct_contribution_count.to_i
+  else
+    0
+  end
+end
+
+def indirect_contributor
+  indirect_contributor_search_args.inject([]) do |cards, search_args|
+    cards + Card.search(search_args)
+  end
+end
+
+def calculate_indirect_contribution_count
+  indirect_contributor.inject(0) do |total, c_card|
+    more_contributions =
+      case c_card
+      when respond_to?(:contribution_count)
```

Comment:
This doesn't work. It's equivalent to `c_card === respond_to?(:contribution_count)` which is always false.